### DISCUSSION TITLE
[Question] Test lifecycle hooks

### DISCUSSION THREAD

**Discussion by iMusicJJ** [livewire/livewire#3479] [1. Help]
Hi,

I've been going over both the testing docs and the lifecycle docs, but I cannot seem to find a way to test lifecycle hooks.

I've got a hook that reads an order number and populates an address from that.

It seems I need a way to tell Livewire to process, since this won't work:

```php
Livewire::test(Create::class)
                ->set('orderNumber', $order->id)
                ->assertSet('address.name', $order->name);
```

With a lifecycle hook that listens for updatedOrderNumber. (It works in practice, but I'd like to write a thorough test too).

I've tried to put a `->call('render')` inbetween set and asset, as well as calling the lifecyclehook in as `->call('updatedOrderNumber')`, but also no dice.

**Comment by joshhanley** [comment]
@iMusicJJ can you share what your component class?

That test should run as you are hoping to, so maybe something else is going on.

**Reply by iMusicJJ** [reply]
Hi, I am a dork.

I took over the test from a colleague; The test was named after one component but was testing a different component.

Apologies

**Reply by joshhanley** [reply]
@iMusicJJ ah no worries! Glad you worked it out 🙂
